### ISSUE TITLE
Cyclomatic Complexity calculation error

### ISSUE THREAD

**Issue by SarthakKumar1997** [rubik/argon#35]
Hi!

I've been using Argon for a while now and it seems to not detect guards.

Here's some code I shamelessly picked out of stackoverflow.

```
nthElement :: [a] -> Int -> Maybe a 
nthElement [] a = Nothing
nthElement (x:xs) a | a <= 0 = Nothing
                    | a == 1 = Just x
                    | a > 1 = nthElement xs (a-1)

nthElementIf :: [a] -> Int -> Maybe a
nthElementIf [] a = Nothing
nthElementIf (x:xs) a = if a <= 1
                        then if a <= 0 
                             then Nothing
                             else Just x -- a == 1
                        else nthElementIf xs (a-1)                           

nthElementCases :: [a] -> Int -> Maybe a
nthElementCases [] a = Nothing
nthElementCases (x:xs) a = case a <= 0 of
                             True -> Nothing
                             False -> case a == 1 of
                                        True -> Just x
                                        False -> nthElementCases xs (a-1)
```

Argon computes a cyclomatic complexity of 4 for `nthElementCases` and 4 for `nthElementIf`, but just 2 for `nthElement`.

I've tried this with a Fibonacci function I wrote, and I get the same problem.

Is it that guards are deliberately not supported, or that there is a bug in the code?

**Comment by rubik** [comment]
Hi Sarthak, it's true that support for guards is lacking. At the moment I have no time to resume the development of Argon, but I'd be happy to review any PRs of yours if you want to contribute.

**Comment by SarthakKumar1997** [comment]
Hey, have solved this issue, but it's in argon-0.4.1.0 and not the latest version of the argon package (for some reason this wasn't building for me). 

Will submit a PR soon :)

**Comment by rubik** [comment]
@SarthakKumar1997 Thank you, I would greatly appreciate that!

**Comment by SarthakKumar1997** [comment]
As this has been merged (#36) I'm closing this issue @rubik. Thanks for allowing me to contribute to Argon!

**Comment by rubik** [comment]
@SarthakKumar1997 You are welcome. I don't have much free time to devote to Argon, so I really appreciate contributions!
